experimental/devices/inky/inky.go

```diff
@@ -0,0 +1,352 @@
+	if dstRect != d.Bounds() {
+		return fmt.Errorf("Partial update not supported")
+	}
+
+	if src.Bounds() != d.Bounds() {
+		return fmt.Errorf("Image must be the same size as bounds: %v", d.Bounds())
+	}
+
+	b := src.Bounds()
+	// Black/white pixels.
+	white := make([]bool, rows*cols)
+	// Red/Transparent pixels.
+	red := make([]bool, rows*cols)
+	for x := b.Min.X; x < b.Max.X; x++ {
+		for y := b.Min.Y; y < b.Max.Y; y++ {
+			i := x*cols + y
+			srcX := x
+			srcY := b.Max.Y - y - 1
+			r, g, b, _ := d.ColorModel().Convert(src.At(srcX, srcY)).RGBA()
+			if r >= 0x8000 && g >= 0x8000 && b >= 0x8000 {
+				white[i] = true
+			} else if r >= 0x8000 {
+				// Red pixels also need white behind them.
+				white[i] = true
+				red[i] = true
+			}
+		}
+	}
+
+	bufA, _ := pack(white)
+	bufB, _ := pack(red)
+	return d.update(borderColor[d.border], bufA, bufB)
+}
+
+func (d *Dev) update(border byte, black []byte, red []byte) error {
+	if err := d.reset(); err != nil {
+		return err
+	}
+
+	if err := d.sendCommand(0x74, []byte{0x54}); err != nil { // Set Analog Block Control.
+		return err
+	}
+	if err := d.sendCommand(0x7e, []byte{0x3b}); err != nil { // Set Digital Block Control.
+		return err
+	}
+
+	r := make([]byte, 3)
+	binary.LittleEndian.PutUint16(r, rows)
+	if err := d.sendCommand(0x01, r); err != nil { // Gate setting
+		return err
+	}
+
+	init := []struct {
+		cmd  byte
+		data []byte
+	}{
+		{0x03, []byte{0x10, 0x01}}, // Gate Driving Voltage.
+		{0x3a, []byte{0x07}},       // Dummy line period
+		{0x3b, []byte{0x04}},       // Gate line width
+		{0x11, []byte{0x03}},       // Data entry mode setting 0x03 = X/Y increment
+		{0x04, nil},                // Power on
+		{0x2c, []byte{0x3c}},       // VCOM Register, 0x3c = -1.5v?
+		{0x3c, []byte{0x00}},
+		{0x3c, []byte{byte(border)}}, // Border colour
+	}
+
+	for _, c := range init {
+		if err := d.sendCommand(c.cmd, c.data); err != nil {
+			return err
+		}
+	}
+
+	switch d.color {
+	case Black:
+		if err := d.sendCommand(0x32, blackLUT[:]); err != nil {
+			return err
+		}
+	case Red:
+		if err := d.sendCommand(0x32, redLUT[:]); err != nil {
+			return err
+		}
+	case Yellow:
+		if err := d.sendCommand(0x04, []byte{0x07}); err != nil { // Set voltage of VSH and VSL.
+			return err
+		}
+		if err := d.sendCommand(0x32, yellowLUT[:]); err != nil {
+			return err
+		}
+	}
+
+	h := make([]byte, 4)
+	binary.LittleEndian.PutUint16(h[2:], rows)
+	write := []struct {
+		cmd  byte
+		data []byte
+	}{
+		{0x44, []byte{0x00, cols/8 - 1}}, // Set RAM X Start/End
+		{0x45, h},                        // Set RAM Y Start/End
+		{0x43, []byte{0x00}},
+
+		{0x4e, []byte{0x00}},
+		{0x4f, []byte{0x00, 0x00}},
+		{0x24, black},
+
+		{0x43, []byte{0x00}},
+		{0x4f, []byte{0x00, 0x00}},
+		{0x26, red},
+
+		{0x22, []byte{0xc7}},
+	}
+
+	for _, c := range write {
+		if err := d.sendCommand(c.cmd, c.data); err != nil {
+			return err
+		}
+	}
+
+	d.busy.In(gpio.PullUp, gpio.FallingEdge)
```

Comment:
I'll commit as soon as gohci test passes but please do the same verification for this one in a follow up.

That said in practice I'm not sure it's really necessary to reset the edge detection here, you could do something like

```
// Ensure any buffered edge is flushed.
d.busy.WaitForEdge(0)

(do initialization)

if !d.busy.WaitForEdge(10*time.Second) {
  return errors.New("failed to initialize")
}
```